libpoolprotocols/getwork/EthGetworkClient.cpp

```diff
@@ -22,7 +22,7 @@ EthGetworkClient::~EthGetworkClient()
 void EthGetworkClient::connect()
 {
 	if (m_connection_changed) {
-		p_client = new ::JsonrpcGetwork(new jsonrpc::HttpClient(m_host));
+		p_client = new ::JsonrpcGetwork(new jsonrpc::HttpClient(m_conn.Host()));
```

Comment:
i'm quite sure this is wrong. 
You must give the complete uri here.

Reply:
Ah, I see. That's why they got away with stuffing host:port into m_farmURL.

Reply:
Does json require a scheme? Or will just host:port do?

Reply:
full URI.. 1:1 what the user specified at ``-F``  or now ``-P``  No parsing necessary. Since jsonrpc::HttpClient uses curl within, which does the parsing itself.